addons/im_livechat/static/src/core/thread_model_patch.js

```diff
@@ -36,8 +36,8 @@ patch(Thread.prototype, "im_livechat", {
         if (!this.correspondent.is_public && this.correspondent.country) {
             return `${this.getMemberName(this.correspondent)} (${this.correspondent.country.name})`;
         }
-        if (this.anonymous_country) {
-            return `${this.getMemberName(this.correspondent)} (${this.anonymous_country.name})`;
+        if (this.channel?.anonymous_country) {
+            return `${this.getMemberName(this.correspondent)} (${this.channel.anonymous_country.name})`;
```

Comment:
Better let `insert`/`update` in `thread_service` handle insert of new field `anonymous_country`.
`channel` field on thread model is deprecated, and we want to remove it some time in the future. Better not adding reliance on this field ;)